===================================================================
RCS file: /cvsroot/gaim/gaim/src/protocols/msn/userlist.c,v
retrieving revision 1.16.2.10
retrieving revision 1.16.2.11
diff -u -r1.16.2.10 -r1.16.2.11
--- gaim/gaim/src/protocols/msn/userlist.c	2005/03/17 05:17:04	1.16.2.10
+++ gaim/gaim/src/protocols/msn/userlist.c	2005/07/08 04:16:17	1.16.2.11
@@ -622,6 +622,15 @@
 
 	group_id = -1;
 
+	if (!gaim_email_is_valid(who))
+	{
+		char *str = g_strdup_printf(_("Unable to add \"%s\"."), who);
+		gaim_notify_error(NULL, NULL, str,
+						  _("The screen name specified is invalid."));
+		g_free(str);
+		return;
+	}
+
 	if (group_name != NULL)
 	{
 		group_id = msn_userlist_find_group_id(userlist, group_name);
